test/test_rpc.py

```diff
@@ -385,5 +385,33 @@ def test_stress_light_rpc(self):
     def test_stress_heavy_rpc(self):
         self._stress_test_rpc(heavy_rpc, repeat=20, args=(torch.ones(100, 100),))
 
-if __name__ == "__main__":
+    @_wrap_with_rpc
+    def test_builtin_remote_ret(self):
+        n = self.rank + 1
+        dst_rank = n % self.world_size
+        rref = dist.remote('worker{}'.format(dst_rank), torch.add,
+                           args=(torch.ones(n, n), torch.ones(n, n)))
+        self.assertEqual(rref.to_here(), torch.ones(n, n) * 2)
+
+    @_wrap_with_rpc
+    def test_multi_builtin_remote_ret(self):
+        m = 10
+        n = self.rank + 1
+        dst_rank = n % self.world_size
+        rrefs = []
+        expected = []
+        for i in range(m):
+            n = n + i
+            rrefs.append(dist.remote(
+                'worker{}'.format(dst_rank),
+                torch.add,
+                args=(torch.ones(n, n), torch.ones(n, n))
+            ))
+            expected.append(torch.ones(n, n) * 2)
+
+        for i in range(m):
+            self.assertEqual(rrefs[i].to_here(), expected[i])
```

Comment:
We need some more tests here:

1. Verify all RRefs are cleaned up properly from RRefContext in the end.
2. Validate state of RRefContext on each node after we've passed a bunch of RRef's around. Also, validate forks are created appropriately.
3. Add test to verify owner of RRef is populated correctly.
4. Take down the owner of an rref and verify to_here() throws an appropriate exception.

Reply:
> Take down the owner of an rref and verify to_here() throws an appropriate exception.

@pritamdamania87  How do we take down the owner of an RRef? Are we going to expose test-only APIs for this?